crates/red_knot_module_resolver/src/path.rs

```diff
@@ -0,0 +1,693 @@
+    #[inline]
+    fn from(value: &'a ModuleResolutionPath) -> Self {
+        match value {
+            ModuleResolutionPath::Extra(ExtraPathBuf(path)) => {
+                ModuleResolutionPathRef::extra_unchecked(path)
+            }
+            ModuleResolutionPath::FirstParty(FirstPartyPathBuf(path)) => {
+                ModuleResolutionPathRef::first_party_unchecked(path)
+            }
+            ModuleResolutionPath::StandardLibrary(StandardLibraryPathBuf(path)) => {
+                ModuleResolutionPathRef::standard_library_unchecked(path)
+            }
+            ModuleResolutionPath::SitePackages(SitePackagesPathBuf(path)) => {
+                ModuleResolutionPathRef::site_packages_unchecked(path)
+            }
+        }
+    }
+}
+
+impl<'a> AsRef<FileSystemPath> for ModuleResolutionPathRef<'a> {
+    #[inline]
+    fn as_ref(&self) -> &FileSystemPath {
+        self.as_file_system_path()
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPath> for ModuleResolutionPathRef<'a> {
+    fn eq(&self, other: &ModuleResolutionPath) -> bool {
+        match (self, other) {
+            (
+                ModuleResolutionPathRef::Extra(ExtraPath(self_path)),
+                ModuleResolutionPath::Extra(ExtraPathBuf(other_path)),
+            )
+            | (
+                ModuleResolutionPathRef::FirstParty(FirstPartyPath(self_path)),
+                ModuleResolutionPath::FirstParty(FirstPartyPathBuf(other_path)),
+            )
+            | (
+                ModuleResolutionPathRef::StandardLibrary(StandardLibraryPath(self_path)),
+                ModuleResolutionPath::StandardLibrary(StandardLibraryPathBuf(other_path)),
+            )
+            | (
+                ModuleResolutionPathRef::SitePackages(SitePackagesPath(self_path)),
+                ModuleResolutionPath::SitePackages(SitePackagesPathBuf(other_path)),
+            ) => *self_path == **other_path,
+            _ => false,
+        }
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPathRef<'a>> for ModuleResolutionPath {
+    fn eq(&self, other: &ModuleResolutionPathRef<'a>) -> bool {
+        other.eq(self)
+    }
+}
+
+impl<'a> PartialEq<FileSystemPath> for ModuleResolutionPathRef<'a> {
+    fn eq(&self, other: &FileSystemPath) -> bool {
+        self.as_file_system_path() == other
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPathRef<'a>> for FileSystemPath {
+    fn eq(&self, other: &ModuleResolutionPathRef<'a>) -> bool {
+        self == other.as_file_system_path()
+    }
+}
+
+impl<'a> PartialEq<VfsPath> for ModuleResolutionPathRef<'a> {
+    fn eq(&self, other: &VfsPath) -> bool {
+        let VfsPath::FileSystem(other) = other else {
+            return false;
+        };
+        self.as_file_system_path() == &**other
+    }
+}
+
+impl<'a> PartialEq<ModuleResolutionPathRef<'a>> for VfsPath {
+    fn eq(&self, other: &ModuleResolutionPathRef<'a>) -> bool {
+        other.eq(self)
+    }
+}
```

Comment:
The `Eq` implementation here are unused and I'm not quiet sure if an equal implementation that doesn't take the module search path variant into account matches what I would expect. 

Let's also remove all other trait implementations that are currently unused. We can always add them once needed.